src/fs/private/tree.ts

```diff
@@ -40,35 +28,42 @@ export class PrivateTree extends PublicTree {
     return obj.putEncrypted !== undefined
   }
 
-  static async empty(version: SemVer = semver.latest, key?: string): Promise<PrivateTree> {
+  static async empty(version: SemVer, key?: string): Promise<PrivateTree> {
     const keyStr = key ? key : await keystore.genKeyStr()
-    return new PrivateTree({}, version, keyStr, {})
+    return new PrivateTree({}, keyStr, {
+      ...header.empty(),
+      version,
+    })
   }
 
-  static async fromCID(_cid: CID): Promise<PublicTree> {
-    throw new Error("This is a private node. Use PrivateNode.fromCIDEncrypted")
-  }
-
-  static async fromCIDWithKey(cid: CID, parentKey: string): Promise<PrivateTree> {
```

Comment:
I'm not sure about this design choice, so I'm eager to hear feedback on it.

I ditched the separate `fromCIDWithKey` function in favor of an optional key param on `fromCID`. 
This makes writing functions that abstract over private and public trees much easier. We still throw an error if a parentKey isn't passed into `fromCID`, but we lose the type safety (and have an `undefined` which I've started trying to avoid).

Other options:
- Go back to the old way. Do runtime type checks when abstracting over private/public
- change `key?: string` to `key: Maybe<string>` which gives extra type safety (you have to intentionally pass in `null` but leads to the awkward situation where to instantiate a public tree, you have to call `PublicTree.fromCID(cid, null)`

Thoughts?